core/chains/evm/config/v2/chain_scoped_gas_estimator.go

```diff
@@ -1,16 +1,37 @@
 package v2
 
 import (
+	gethcommon "github.com/ethereum/go-ethereum/common"
+
 	"github.com/smartcontractkit/chainlink/v2/core/assets"
 	"github.com/smartcontractkit/chainlink/v2/core/chains/evm/config"
 )
 
 type gasEstimatorConfig struct {
 	c                       GasEstimator
+	k                       KeySpecificConfig
 	blockDelay              *uint16
 	transactionsMaxInFlight *uint32
 }
 
+func (g *gasEstimatorConfig) KeySpecificMaxPrice(addr gethcommon.Address) *assets.Wei {
```

Comment:
Maybe we should consider naming this a little differently?
```suggestion
func (g *gasEstimatorConfig) MaxPriceKey(addr gethcommon.Address) *assets.Wei {
```
```suggestion
func (g *gasEstimatorConfig) MaxPriceByKey(addr gethcommon.Address) *assets.Wei {
```
```suggestion
func (g *gasEstimatorConfig) MaxPriceForKey(addr gethcommon.Address) *assets.Wei {
```